wire/internal/wire/wire_test.go

```diff
@@ -59,116 +61,108 @@ func TestWire(t *testing.T) {
 	}
 	wd := filepath.Join(magicGOPATH(), "src")
 
-	t.Run("Generate", func(t *testing.T) {
-		if _, err := os.Stat(filepath.Join(build.Default.GOROOT, "bin", "go")); err != nil {
-			t.Skip("go toolchain not available:", err)
-		}
-		for _, test := range tests {
-			test := test
-			t.Run(test.name, func(t *testing.T) {
-				t.Parallel()
-
-				// Run Wire from a fake build context.
-				bctx := test.buildContext()
-				gen, errs := Generate(bctx, wd, test.pkg)
-				if len(gen) > 0 {
-					defer t.Logf("wire_gen.go:\n%s", gen)
+	if _, err := os.Stat(filepath.Join(build.Default.GOROOT, "bin", "go")); err != nil {
+		t.Skip("go toolchain not available:", err)
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			// Run Wire from a fake build context.
+			bctx := test.buildContext()
+			gen, errs := Generate(bctx, wd, test.pkg)
+			if len(gen) > 0 {
+				defer t.Logf("wire_gen.go:\n%s", gen)
+			}
+			if len(errs) > 0 {
+				for _, e := range errs {
+					t.Log(e)
 				}
-				if len(errs) > 0 {
-					for _, e := range errs {
-						t.Log(e)
-					}
-					if !test.wantError {
-						t.Fatal("Did not expect errors.")
-					}
-					for _, s := range test.wantErrorStrings {
-						if !errorListContains(errs, s) {
-							t.Errorf("Errors did not contain %q", s)
-						}
-					}
-					return
+				if !test.wantError {
+					t.Fatal("Did not expect errors.")
 				}
-				if len(errs) == 0 && test.wantError {
-					t.Fatal("wirego succeeded; want error")
+				for _, s := range test.wantErrorStrings {
+					if !errorListContains(errs, s) {
+						t.Errorf("Errors did not contain %q", s)
+					}
 				}
+				return
+			}
+			if len(errs) == 0 && test.wantError {
+				t.Fatal("wirego succeeded; want error")
+			}
 
-				// Find the absolute import path, since test.pkg may be a relative
-				// import path.
-				genPkg, err := bctx.Import(test.pkg, wd, build.FindOnly)
-				if err != nil {
-					t.Fatal(err)
+			goldenDir := filepath.Join("testdata", test.name, ".golden")
```

Comment:
WDYT about loading the "golden" file in loadTestData below? And updating the comments for that function to include the new golden file.

Reply:
There's also an existing "parseGoldenFile" func that should probably be renamed, as it's now a bit confusing since it's not about the golden file you're adding here.

Reply:
I updated loadTestData as suggested, and moved the out.txt files into the .golden dirs. I didn't rename the `parseGoldenOutput` function since I think the name is accurate, but I renamed some fields for greater clarity.